art/attacks/evasion/hop_skip_jump.py

```diff
@@ -188,8 +234,42 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
 
         return x_adv
 
+    @staticmethod
+    def _get_mask(x: np.ndarray, classifier_mixin: bool = True, **kwargs) -> np.ndarray:
+        """
+        Get the mask from the kwargs.
+
+        :param x: An array with the original inputs.
+        :param classifier_mixin: Whether the estimator is of type `ClassifierMixin`.
+        :param mask: An array with a mask to be applied to the adversarial perturbations. Shape needs to be
+                     broadcastable to the shape of x. Any features for which the mask is zero will not be adversarially
+                     perturbed.
+        :type mask: `np.ndarray`
+        :return: The mask.
+        """
+        mask = kwargs.get("mask")
+
+        if mask is not None:
+            if classifier_mixin:
+                # Ensure the mask is broadcastable
+                if len(mask.shape) > len(x.shape) or mask.shape != x.shape[-len(mask.shape) :]:
+                    raise ValueError("Mask shape must be broadcastable to input shape.")
+
+            else:
+                raise ValueError("Mask is only supported for classification.")
```

Comment:
I think the mask should also work for tasks other than classification without any additional modifications, or?

Reply:
Solved.